CODE_REVIEW.md

```diff
@@ -0,0 +1,16 @@
+# Improvements:
+
+* Website is not responsive for mobile devices, made it responsive.
+* Private member variable ‘http' is never reassigned, marked it as 'readonly’.
+* Filtering reading list API (/api/reading-list) response and passing only the list with book items, thereby avoiding any empty objects.
+* We can implement caching our HTTP requests to avoid hitting same endpoints multiple times.
```

Comment:
When user searches for a keyword, we can use a service to cache the results of bookk search api (**/api/books/search?q=${action.term}**) response and retrieve the data from cache if he searches for same keyword again thereby avoid hitting same endpoint multiple times. Since reading the data from cache is extremely fast, this improves overall performance of the application.